rfc/plugins.md

```diff
@@ -0,0 +1,300 @@
+// plugin framwework which runtimes are acceptable. This
+// annotation would additionally be used for other metadata
+// Spinnaker needs to know atop whatever PF4J offers out of
+// the box.
+@SpinnakerExtension(drivers = [
+  ExtensionDriver.LOCAL, 
+	ExtensionDriver.REMOTE
+])
+class PipelineStageExtension : ExtensionPoint {
+	// ...
+}
+```
+
+It is highly recommended to read the documentation of PF4J to fully understand the scope of this proposal.
+
+### Front50 as the Source of Truth
+Operators will need the ability to understand, at a glance, what plugins are installed and enabled or disabled, their versions, as well as any other metadata about a plugin. Whether plugins are implemented as in-process or not, Front50 should be used to track and expose a single source of truth of what plugins are installed in any particular service, as well as which ones are enabled or disabled.
+
+The Front50 plugin registry will also be exposed through Gate, and can be consumed by Deck to load the correct plugins on application startup.
+
+Plugins can also be enabled or disabled at runtime without redeploying a service. This flag is sourced from Front50, which all services will be listening for change events from. Whether this change event is published or polled for initially is unknown.
+
+Halyard configs can be used to inform services of their _initial state_. When a service starts up, it will notify Front50 of what in-process plugins are installed, however Front50 will be responsible for determining the enabled state.
+
+Using Front50 as a source of truth will require extension of PF4J.
+
+```proto
+syntax = "proto3";
+
+// PluginManifest is used during registration of a plugin
+message PluginManifest {
+	// The name of the plugin, which is used as a fully 
+	// qualified identifier, including a namespace and 
+	// plugin name, ex: "netflix/fast-properties"
+	string name = 1;
+
+	// A short description of the plugin
+	// ex: "Provides rollout safety of runtime application 
+	// config via pipelines"
+	string description = 3;
+
+	// The name of the plugin author, ex: "Netflix"
+	string provider_name = 4;
+
+	// The semver plugin version, ex: "1.0.0"
+	string version = 5;
+
+	// A list of plugin dependencies. Initially this will be
+	// informational only. ex: "netflix/somedep>=1.x"
+	repeated string dependencies = 6;
+
+	// required_versions is a repeated string of service version	// requirements. e.g. "clouddriver >= 5.33.0"
+	repeated string required_versions = 7;
+
+	// A list of extension points that the plugin implements.
+	// e.g. "pipelineStage"
+	repeated string provided_capabilities = 8;
+
+	// Whether or not the plugin is enabled. An unset value 
+	// should be considered as true.
+	boolean enabled = 9;
+}
+```
+
+#### Sending Plugin Configuration to Deck
+
+While Front50 will be the source of truth for which plugins are enabled, we can pass that data via settings.js today and transition to using Front50 when it is ready.
```

Comment:
> we can pass that data via settings.js today
Maybe rewrite as "we can pass that data via Deck's configuration file (`settings.js`) today" in case readers are unfamiliar with Deck's internals?